alacritty/src/display.rs

```diff
@@ -498,8 +517,29 @@ impl Display {
             });
         }
 
+        #[cfg(not(any(target_os = "macos", windows)))]
+        self.request_frame(&self.window);
+
         self.window.swap_buffers();
     }
+
+    /// Requst a new frame for a window on Wayland.
+    #[inline]
+    #[cfg(not(any(target_os = "macos", windows)))]
+    fn request_frame(&self, window: &Window) {
+        // Register wayland frame callback
+        if let Some(surface) = window.wayland_surface() {
```

Comment:
Could pull out the surface here in a match to reduce indentation. But the method's so short that it might not be worth it to add the extra two lines required to do so.

I'd say it's right on the edge where either thing would work fine, so I'll leave it up to you to decide.